docs/source/data/kedro_dataset_factories.md

```diff
@@ -225,3 +225,145 @@ You can use dataset factories to define a catch-all pattern which will overwrite
 ```
 Kedro will now treat all the datasets mentioned in your project's pipelines that do not appear as specific patterns or explicit entries in your catalog
 as `pandas.CSVDataSet`.
+
+## CLI commands for dataset factories
+
+To manage your dataset factories, two new commands have been added to the Kedro CLI: `kedro catalog rank` (0.18.12) and `kedro catalog resolve` (0.18.13).
+
+#### Using `kedro catalog rank`
+
+This command outputs a list of all dataset factories in the catalog, ranked in the order by which pipeline datasets are matched against them. This ordering is determined by considering the following criteria:
+
+1. The number of non-placeholder characters in the pattern
+2. The number of placeholders in the pattern
+3. Alphabetic ordering
+
+Consider a catalog file with the following patterns:
+
+```yaml
```

Comment:
I'm leaning towards suggesting that we put the code blocks in the click to open <summary><details> tags. WDYT?

Reply:
Sounds good, changed in [507c1d1](https://github.com/kedro-org/kedro/pull/2935/commits/507c1d142a03ef7c667d1b43b8eb39a853dd496e). I left the output of `kedro catalog rank` as it was as it's rather short and isn't technically a code snippet, but happy to make it uniform too